connectors/sources/redis.py

```diff
@@ -0,0 +1,280 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+import json
+from contextlib import asynccontextmanager
+from enum import Enum
+
+import redis.asyncio as redis
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource, ConfigurableFieldValueError
+from connectors.utils import hash_id, iso_utc
+
+PAGE_SIZE = 1000
+MAX_POOL_SIZE = 10
+
+
+class KeyType(Enum):
+    HASH = "hash"
+    STREAM = "stream"
+    LIST = "list"
+    SET = "set"
+    ZSET = "zset"
+    JSON = "ReJSON-RL"
+
+
+class RedisClient:
+    """Redis client to handle method calls made to Redis"""
+
+    def __init__(self, configuration):
+        self.configuration = configuration
+        self._logger = logger
+        self.host = self.configuration["host"]
+        self.port = self.configuration["port"]
+        self.database = self.configuration["database"]
+        self.username = self.configuration["username"]
+        self.password = self.configuration["password"]
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+
+    @asynccontextmanager
+    async def client(self):
+        try:
+            pool = redis.ConnectionPool.from_url(
+                f"redis://{self.username}:{self.password}@{self.host}:{self.port}",
+                decode_responses=True,
+                max_connections=MAX_POOL_SIZE,
+            )
+            client = redis.Redis(connection_pool=pool)
+            yield client
+        finally:
+            await client.aclose()  # pyright: ignore
+
+    async def validate_database(self, db, redis_client):
+        try:
+            await redis_client.execute_command("SELECT", db)
+            return await redis_client.ping()
+        except Exception as exception:
+            self._logger.warning(f"Database {db} not found. Error: {exception}")
```

Comment:
Should this method explicitly return `Ture/False`?

Reply:
yes we need to mention it explicitly because on the basis of that we are validating valid and invalid databases.